content/terraform-docs-common/docs/cloud-docs/api-docs/hold-your-own-key/aws-oidc-configurations.mdx

```diff
@@ -0,0 +1,166 @@
+---
+page_title: AWS specific /oidc-configuration API reference for HCP Terraform
+description: >-
+ Use the HCP Terraform API's `/oidc-configuration` endpoint to create and view your AWS OIDC configurations for Hold Your Own Key (HYOK) in HCP Terraform.
+---
+
+[200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200
+
+[201]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201
+
+[202]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202
+
+[204]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204
+
+[400]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
+
+[401]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401
+
+[403]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
+
+[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
+
+[409]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409
+
+[412]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412
+
+[422]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
+
+[429]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429
+
+[500]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500
+
+[504]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504
+
+[JSON API document]: /terraform/cloud-docs/api-docs#json-api-documents
+
+[JSON API error object]: https://jsonapi.org/format/#error-objects
+
+# AWS OIDC Configuration API reference
+
```

Comment:
It could be worth calling out that these configurations are only available on HCP Terraform (not TFE), and that they require Premium Tier

```suggestion

Hold Your Own Key (HYOK) is only available on the [HCP Terraform Premium plan](https://www.hashicorp.com/en/pricing?tab=terraform. This includes the management of OIDC configurations.

```